nix/hosts/webforge/web-landing-page.nix

```diff
@@ -0,0 +1,91 @@
+    "d /var/www/live 0775 bot-www nginx"
+    "d /var/www/preview 0775 bot-www nginx"
+  ];
+
+  services.nginx = {
+    # The service and the firewall should already be configured elsewhere - see configuration.nix
+    # But we need to enable it here (again) to evaluate the `nginx` group in the lines above.
+    enable = true;
+
+    # Configure the virtualhosts to serve the pages
+    # TODO: Replace 87b59b92.nip.io by tahoe-lafs.org below when ready - trac#4162
+    virtualHosts = {
+      # Define a live site to serve the content generated from the main branch
+      # See https://github.com/tahoe-lafs/web-landing-page
+      "live.www.87b59b92.nip.io" = {
+        enableACME = true;
+        forceSSL = true;
+        root = "/var/www/live";
+        serverAliases = [
+          "www.87b59b92.nip.io"
+          "87b59b92.nip.io"
+        ];
+        extraConfig = ''
+          # Redirect to the main domain
+          if ($http_host != 87b59b92.nip.io) {
+            rewrite ^(.*)$ https://87b59b92.nip.io$1 redirect;
+          }
+          # Redirect unmigrated pages to the legacy site:
+          # - all trac projects - tahoe-lafs is pending for Forgejo, others are mostly stalled
+          # - downloads - still used to publish new releases
+          # - pipermail - still serving some mailing archive (not older than 02-Dec-2021) before lists.tahoe-lafs.org
+          # - hacktahoelafs - to be migrated as a simple blog post or a special page
+          # - user home dirs (e.g. ~trac, ~warner and ~zooko) - still holds files referred from some other places
+          rewrite (trac|downloads|pipermail|hacktahoelafs|~[^/]+)(|/.*)$ https://legacy.87b59b92.nip.io/$1$2 redirect;
+        '';
+      };
+
+      # Define a preview site to serve the content generated from per pull-requests
+      # See https://github.com/tahoe-lafs/web-landing-page/pulls
+      "preview.www.87b59b92.nip.io" = {
+        enableACME = true;
+        forceSSL = true;
+        root = "/var/www/preview";
+      };
+
+      # Proxy the legacy site under a separate hostname to allow (read) access if needed
+      "legacy.87b59b92.nip.io" = {
+        enableACME = true;
+        forceSSL = true;
+        locations = {
+          # Proxy all requests to legacy linode server
+          "/" = {
+            proxyPass = "https://74.207.252.227/";
+          };
+          # Only because the original file do not exist uncompressed!
+          # And the proxy fails to get the compressed one, unlike a browser?
+          # FIXME: (re-)create the missing ~trac/LAFS.svg on the legacy server
+          # Or fix the proxy (headers?) to access the compressed file?
+          "/~trac/LAFS.svg" = {
+            proxyPass = "https://74.207.252.227/~trac/LAFS.svg.gz";
+            extraConfig = ''
+              proxy_hide_header Content-Type;
+              add_header        Content-Encoding gzip;
+              add_header        Content-Type image/svg+xml;
+            '';
+          };
```

Comment:
Not sure I understand - this doesn't look right to me.  We'd have to have one such block for every file we find where there's something wrong with the compression?  (on-disk but - rightly so- not also on-wire?)

Is this because the trac server serves pre-compressed files when the client supports gzip compression ([like so](https://stackoverflow.com/a/9158330/1181063))?

Looks like our proxy discard the request's `Accept` header?

Maybe [`proxy_pass_request_headers`](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_pass_request_headers) (https://stackoverflow.com/a/72001522/1181063) helps here?

Reply:
I've re-tested a few things and was not able to reproduce the problem at first! :cry: 

But then I've figured that the recent merge of #51 had changed something in regard with the `recommendedProxySettings`.
And indeed, I've isolated one parameter which does interfere with the behavior I've initially observed: `proxy_http_version 1.1;`

The good news is that this whole exception bloc can go.
But I'm not sure if we should still add `proxy_pass_request_headers on` to be safe!?

Reply:
Arf, sorry @hacklschorsch : I've my answer: `proxy_pass_request_headers on;` does not solve this problem.
But `proxy_http_version 1.1;`, implicitly configured with `recommendedProxySettings = true;` does.
I've removed this exception entirely.